Maple2Storage/Types/Metadata/RecipeMetadata.cs

```diff
@@ -0,0 +1,180 @@
+
+        protected bool Equals(RecipeMetadata other)
+        {
+            return false;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null)
+                return false;
+            if (ReferenceEquals(this, obj))
+                return true;
+            if (obj.GetType() != GetType())
+                return false;
+            return Equals((RecipeMetadata) obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(Id, MasteryType, RequireItem1, RewardItem1);
+        }
+
+        public List<RecipeItem> GetIngredients()
+        {
+            List<RecipeItem> result = new List<RecipeItem>();
+            List<string> requiredItems = new List<string>
+            {
+                RequireItem1,
+                RequireItem2,
+                RequireItem3,
+                RequireItem4,
+                RequireItem5
+            };
+
+            foreach (string requiredItem in requiredItems)
+            {
+                if (!string.IsNullOrEmpty(requiredItem))
+                {
+                    int[] split = SplitStringIntoInts(requiredItem);
+                    result.Add(new RecipeItem
+                    {
+                        Id = split[0],
+                        Amount = split[2]
+                    });
+                }
+            }
+
+            return result;
+        }
+
+        public List<RecipeItem> GetResult()
+        {
+            List<RecipeItem> result = new List<RecipeItem>();
+            List<string> rewardItems = new List<string>
+            {
+                RewardItem1,
+                RewardItem2,
+                RewardItem3,
+                RewardItem4,
+                RewardItem5
+            };
+
+            foreach (string rewardItem in rewardItems)
+            {
+                if (!string.IsNullOrEmpty(rewardItem))
+                {
+                    int[] split = SplitStringIntoInts(rewardItem);
+                    result.Add(new RecipeItem
+                    {
+                        Id = split[0],
+                        Rarity = split[1],
+                        Amount = split[2]
+                    });
+                }
+            }
+
+            return result;
+        }
+
+        public static bool operator ==(RecipeMetadata left, RecipeMetadata right)
+        {
+            return Equals(left, right);
+        }
+
+        public static bool operator !=(RecipeMetadata left, RecipeMetadata right)
+        {
+            return !Equals(left, right);
+        }
+
+        static int[] SplitStringIntoInts(string list)
+        {
+            string[] split = list.Split(new char[1] { ',' });
+            List<int> numbers = new List<int>();
+
+            foreach (string n in split)
+            {
+                if (int.TryParse(n, out int parsed))
```

Comment:
Add braces here

Reply:
Also this seems like more a utility function, perhaps this belongs in another file?